### PR TITLE
determine api root from consolepath var on console

### PR DESCRIPTION
**Pull request by wawhal**
### Description

What component does this PR affect? 

- [ ] Server
- [x] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] Community content

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [ ] This change requires a change in the documentation. 
- [ ] I have updated the documentation accordingly.
- [ ] I have added required tests.

### CODE REVIEW

File: console/src/Globals.js [hunk 1]
```diff
@@ -63,8 +63,22 @@ if (
 
 if (globals.consoleMode === SERVER_CONSOLE_MODE) {
   if (globals.nodeEnv !== 'development') {
-    const windowUrl = window.location.protocol + '//' + window.location.host;
-    globals.dataApiUrl = windowUrl;
+    if (window.__env.consolePath) {
```
Comment (praveenweb) [inline console/src/Globals.js]: add this env in helpers/Html.js

**Comment by mastoj** [comment]
Do I need to set any environment variable to get this working? I have exactly this issue. I want to server the console on http://<my host>/something/console

**Comment by shahidhk** [comment]
@mastoj No, this should be taken care of automatically. You just have to put a reverse proxy in front of Hasura and forward requests to Hasura without the URL prefix.

**Comment by mastoj** [comment]
@shahidhk you're completely right, but I couldn't get the nginx proxy to work the way I wanted in combination with ingress on kubernetes. I did however get it to work after switching from nginx to traefik.

**Comment by KloudJack** [comment]
Heads up to anyone trying to serve graphql-engine behind an nginx reverse proxy and a URL prefix:

You need to append a slash in proxy_pass.
```
location /gql-engine {
    proxy_pass http://graphql-engine/;
}
```
Without the trailing slash the graphql-engine will return an error.

**Comment by levitanong** [comment]
Given @KloudJack 's setup with `/api` as the prefix, if you visit `https://your-domain.com/api`, you'll be redirected to `https://your-domain.com/console`.

Visiting `https://your-domain.com/api/console` is a good enough workaround, but I can't help feeling the `/console` redirect should also take the prefix into account.

**Pull request merged**